PWGHF/TableProducer/HFCandidateCreator3Prong.cxx

```diff
@@ -211,6 +211,15 @@ struct HFCandidateCreator3ProngExpressions {
         flag = sign * (1 << DecayType::DPlusToPiKPi);
       }
 
+      // Ds± → K± K∓ π±
+      if (flag == 0) {
+        // Printf("Checking Ds± → K± K∓ π±");
+        indexRec = RecoDecay::getMatchedMCRec(particlesMC, arrayDaughters, pdg::Code::kDs, array{+kKPlus, -kKPlus, +kPiPlus}, true, &sign);
```

Comment:
If I am not wrong, that should be fine because the decay vertex of the phi meson is not reconstructed since it does not travel much from the Ds decay vertex. In addition, I am going to implement a selection on the invariant mass of the kaon couple in order to ensure that the kaons come from a phi decay. Maybe also @fgrosa can comment on this.

Reply:
We are talking about MC matching here, not reconstruction. If the kaons come from a φ here, the Ds will be labelled as unmatched.

Reply:
Hi @vkucera shouldn't the method look also for resonant decays? If not then it should be modified because we also have resonant decays in the Lc -> pKpi and D+ -> Kpipi for example, that should be considered when asking for the KKpi in the final state. 
For this specific case, I think that as long as the cut on the KK mass is not added at reco level (that I would suggest to add in a second step), all the Ds to KKpi should be considered in the MC (but keep in mind that in some MC productions we might have only Ds --> phi pi --> KKpi).

Reply:
Ah indeed! It should be at least 2

Reply:
Actually, there is the same problem for D+ and Xic+ in this file. @stefanopolitano could you please fix them as well? Thanks a lot!